fvtest/compilertriltest/LongAndAsRotateTest.cpp

```diff
@@ -0,0 +1,494 @@
+
+// This vector consists of all cases where contiguous 1's appear at most once in the binary value
+// ex: 001111000, 1111111, 011110, 1110000, 00001111
+std::vector<uint64_t> lconstInput_zerosAroundOnes =     {   0x0,
+                                                            0x1,                //00000..00001
+                                                            0x00000000FFFFFFFF, //00000..11111
+                                                            0x2,                //00000..00010
+                                                            0x3E,               //00...0111110     
+                                                            0x8000000000000000, //10000000..00
+                                                            0xFFFFFFFF00000000, //111111..0000
+                                                            0x4000000000000000, //010000..0000
+                                                            0x7C00000000000000, //0111100..000
+                                                            0xFFFFFFFFFFFFFFFF, //1111111..111
+                                                            0x00000003C0000000, //0..011110..0
+                                                            0x00007FFFFFFFFFFF,
+                                                            0x0000FFFFFFFFFFFF,
+                                                            0x000000000000FFFF,
+                                                            0xFFFFFFFFFFFE0000,
+                                                            0xFFFFFFFFFFFF0000,
+                                                            0xFFFF000000000000                     
+                                                        };
+
+// this vector is for tests where contiguous zeros only appear once in the binary value.
+// ex: 11100001111, 1000001, 0000000
+std::vector<uint64_t> lconstInput_onesAroundZeros = { 0x0, 0x8000000000000001, 0xE000000000000003, 0xFFFFFFF00FFFFFFF, 0xFFFFFFFEFFFFFFFF, 0xFFFFFFFE7FFFFFFF, 0xF0FFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFF0F, 0XFFFF000000000001, 0x8000FFFFFFFFFFFF };
+
+// These input values cannot be handled by a RISBG instruction
+std::vector<uint64_t> lconstInput_invalidPatterns = {0x4000000000000001, 0xF0F0F0F0F0F0F0F0, 0x8000000000000002};
+
+template <typename VarType, typename ConstType>
+using LongAndAsRotateParamType = std::tuple<VarType, ConstType, ConstType (*) (VarType, ConstType)>;
+
+template <typename VarType, typename ConstType>
+class LongAndAsRotateTest : public ::testing::TestWithParam<LongAndAsRotateParamType<VarType, ConstType>>
+   {
+   public:
+
+   static void SetUpTestCase()
+      {
+      const char *options = "-Xjit:disableTreeSimplification,disableAsyncCompilation,acceptHugeMethods,enableBasicBlockHoisting,omitFramePointer,useIlValidator,paranoidoptcheck";
+      auto initSuccess = initializeJitWithOptions(const_cast<char*>(options));
```

Comment:
Can you open an issue to change the signature of `initializeJitWithOptions` so that this `const_cast` becomes un-necessary?

Reply:
I'm guessing you already got to it here?
https://github.com/eclipse/omr/issues/1916

Reply:
Actually, that's a separate issue entirely

Reply:
Done: https://github.com/eclipse/omr/issues/1946